### PR TITLE
Makes layout and fragment for hmc5883l and mpu6050

### PR DESCRIPTION
**Pull request by asitava1998**
Fixes issue #279 

Changes:
* Sensor QuickView layout is changed to have a list of available Sensors.
* UI for HMC5883L and MPU6050 is created.
* Fragments for HMC5883L and MPU6050 is made.

Note:
* Graph has not been implemented. @akarshan96 said he would do so and some modifications in UI in another PR.

Screenshots for the change:

### CODE REVIEW

**Comment by viveksb007** [comment]
@asitava1998 could you add a screencast for the whole process of detecting sensor and showing real time raw data continuously?
And please resolve the merge conflicts.
Thanks

**Comment by asitava1998** [comment]
@viveksb007  I haven't yet implemented the real time updating. Let me update it then and then prepare the screencast. I think refreshing every 0.5s will be good as it will be proper for readability.

**Comment by asitava1998** [comment]
@viveksb007 Here is the [screencast](https://drive.google.com/file/d/0B3F6wGD_1dcZWHZUbEd4U3NMWlU/view?usp=sharing) refreshing every second. Some parsing of the data is still needed as there are some unnecessary digits in the data. This can be done in another issue.

**Comment by CloudyPadmal** [comment]
Hello @asitava1998 ! Great work! :100: Since we can open new issues for the rest of work, can we remove the WIP label? Then this can be approved and merged! :)

**Comment by asitava1998** [comment]
@CloudyPadmal  Yeah, please go ahead and review it. I removed the WIP label.

File: app/src/main/java/org/fossasia/pslab/fragment/SensorFragmentMPU6050.java [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+
+        new Timer().schedule(new TimerTask() {
+            @Override
+            public void run() {
+                getActivity().runOnUiThread(new Runnable() {
+                    @Override
+                    public void run() {
+                        MPU6050 MPU6050 = null;
+                        try {
+                            MPU6050 = new MPU6050(i2c);
```
Comment (viveksb007) [inline app/src/main/java/org/fossasia/pslab/fragment/SensorFragmentMPU6050.java]: Created an MPU6050 object here also and above too. Timer would execute this TimerTask at specified delay again and again. So object is allocated again and again periodically.
And I don't see any code to handle the Timer when I exit back. Doesn't it give any error when you exit before cancelling the Timer?
And you are reading values on UI thread. Not recommended, would result in lagging UI during chart plot. Currently readback value are less, so its not that visible.
Have a look at AsyncTask to implement this or if you can handle data across threads that okay too.

File: app/src/main/java/org/fossasia/pslab/fragment/SensorFragmentMPU6050.java [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+        SensorFragmentMPU6050 sensorFragmentMPU6050 = new SensorFragmentMPU6050();
+        return sensorFragmentMPU6050;
+    }
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        scienceLab = ScienceLabCommon.scienceLab;
+        i2c = scienceLab.i2c;
```
Comment (viveksb007) [inline app/src/main/java/org/fossasia/pslab/fragment/SensorFragmentMPU6050.java]: sciencelab object is redundant.
```i2c = ScienceLabCommon.sciencelab.i2c``` would do

**Comment by asitava1998** [comment]
@viveksb007 I have implemented the AsyncTask to fetch raw data in the background. I will look into the error in the values displayed and create another issue for the same.

File: app/src/main/java/org/fossasia/pslab/fragment/SensorFragmentHMC5883L.java [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+                }
+            }
+        };
+        new Thread(runnable).start();
+        return view;
+    }
+
+
+    private class SensorDataFetch extends AsyncTask<Void, Void, Void> {
+        HMC5883L HMC5883L = new HMC5883L(i2c);
```
Comment (viveksb007) [inline app/src/main/java/org/fossasia/pslab/fragment/SensorFragmentHMC5883L.java]: whats with initialising ```HMC5883L``` object  again and again?
You initialised it in onCreateView too.
Try making a global ```HMC5883L``` object and initialise it once. And shift code from Line 53-75 to onViewCreated method.

File: app/src/main/java/org/fossasia/pslab/fragment/SensorFragmentHMC5883L.java [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+    public static SensorFragmentHMC5883L newInstance() {
+        SensorFragmentHMC5883L sensorFragmentHMC5883L = new SensorFragmentHMC5883L();
+        return sensorFragmentHMC5883L;
+    }
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        scienceLab = ScienceLabCommon.scienceLab;
+        i2c = scienceLab.i2c;
```
Comment (viveksb007) [inline app/src/main/java/org/fossasia/pslab/fragment/SensorFragmentHMC5883L.java]: I pointed before that ```sciencelab``` object is redundant.
```i2c = ScienceLabCommon.sciencelab.i2c``` would work.

**Comment by viveksb007** [comment]
@asitava1998 try incorporating above suggested changes and we would open another issue to deal with error in sensor values.
Thanks

File: app/src/main/java/org/fossasia/pslab/fragment/SensorFragmentHMC5883L.java (hunk 4 above)
Comment (asitava1998) [inline app/src/main/java/org/fossasia/pslab/fragment/SensorFragmentHMC5883L.java]: Then how is scienceLab.isConnected() supposed to work.

Comment (viveksb007) [inline app/src/main/java/org/fossasia/pslab/fragment/SensorFragmentHMC5883L.java]: well we would reach that state after scanning the sensors that implicitly means that device is connected. But there are cases in which its necessary.
Sorry for that, lets rectify the other one and we are good to merge this PR.

File: app/src/main/java/org/fossasia/pslab/fragment/SensorFragmentHMC5883L.java (hunk 3 above)
Comment (asitava1998) [inline app/src/main/java/org/fossasia/pslab/fragment/SensorFragmentHMC5883L.java]: I made a global `HMC5883L` object and initialized once and shifted the code on the specified lines to `OnCreate` method. However, if the above line 81 is removed, the fragment crashes. Can you identify the reason? I will make the commit accordingly.

Comment (viveksb007) [inline app/src/main/java/org/fossasia/pslab/fragment/SensorFragmentHMC5883L.java]: try passing ```HMC5883L``` object to AsyncTask's contructor. might be issue where both threads try to access at the same time. If it still crashes provide the logs or try to debug it yourself.

**Comment by viveksb007** [comment]
@asitava1998 If it still crashes, we can merge this I would have a look to prevent re-initialisation in a loop.

**Comment by asitava1998** [comment]
@viveksb007 Yes it would be better if we merge and you can have a look. Also passing it to the constructor of AsyncTask did not help. It still crashes.

**Comment by viveksb007** [comment]
Yeah I will look into it. Its most probably due to single object used by 2 threads.

**Comment by asitava1998** [comment]
Yeah. It's working fine right now.

**Pull request merged**